src/lseq/ident.rs

```diff
@@ -94,9 +80,11 @@ impl<A: Actor> IdentGen<A> {
     pub fn alloc(&mut self, p: &Identifier<A>, q: &Identifier<A>) -> Identifier<A> {
         assert!(p != q, "allocation range should be non-empty");
         if q < p {
-            self.alloc(q, p);
+            return self.alloc(q, p);
```

Comment:
this was a bug, we would continue after this if to allocate with `q` still smaller than `p`.